src/shared/hooks/require-auth.hook.ts

```diff
@@ -0,0 +1,20 @@
+import { useAuth } from '#/shared/hooks/auth.hook';
+import { enqueueSnackbar } from 'notistack';
+import { authToastMessages } from '../constants/auth-toast-messages.constant';
+
+export function useRequireAuth() {
+  const { currentUser } = useAuth();
+
+  return (callback: () => void) => {
```

Comment:
Wrap it in the useCallback for ref stability

Reply:
> Wrap it in the useCallback for ref stability

Done